jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java

```diff
@@ -121,14 +121,24 @@ private void addRoute(KubernetesListBuilder listBuilder, ServiceBuilder serviceB
         }
     }
 
-    private static RoutePort createRoutePort(ServiceBuilder serviceBuilder) {
-        RoutePort routePort = null;
-        final Integer serviceTargetPort = getTargetPortToExpose(serviceBuilder);
-        final Integer servicePort = serviceTargetPort != null ? serviceTargetPort : getPortToExpose(serviceBuilder);
-        if (servicePort != null) {
+    private static RoutePort createRoutePort(ServicePort servicePortToExpose) {
+        RoutePort routePort = new RoutePort();
```

Comment:
> nit, maybe rather than creating RoutePort object in two places in the method, we can rely on `RoutePortBuilder` instead and build the final object?

It was a leftover, there's no need to create the RoutePort object twice.